### PR TITLE
Github Actions Fixes

### PR DESCRIPTION
**Pull request by samcat116**
- Removed the `gitleaks` action since it's broke upstream
- Added support to build and push the docker image to the registry
- Started deployment steps, not configured for now

### CODE REVIEW

**Comment by ironchefpython** [comment]
I have no idea what this is, but I approve!

File: .github/workflows/lintandtest.yml [hunk 1]
```diff
@@ -9,16 +13,23 @@ jobs:
...
+      run: |
+        pip install pylint
+        pylint -E *.py
+        
     - name: Test with unittest
-      run: python3 -m unittest
+      run: |
+        python3 -m unittest
+    - name: Build Docker Image
+      run: docker build .
```
Comment (jswny) [inline .github/workflows/lintandtest.yml]: Do we need to name/tag the image here?

Comment (samcat116) [reply .github/workflows/lintandtest.yml]: No this is basically just testing that that it builds, not pushing at this step

File: .github/workflows/lintandtest.yml [hunk 2]
```diff
@@ -9,16 +13,23 @@ jobs:
...
-        python-version: 3.8
+        python-version: '3.8'
+
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
         pip install -r requirements.txt
+
     - name: Lint with pylint
-      run: pylint -E nachomemes
```
[the remark this answers is not in the document]
Comment (samcat116) [reply .github/workflows/lintandtest.yml]: Pushed this fix

**Pull request merged**